pkg/cli/dump.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/cockroachdb/errors"
+	"github.com/spf13/cobra"
+)
+
+// dumpCmd dumps SQL tables.
+var dumpCmd = &cobra.Command{
+	Use:   "dump [options] <database>",
+	Short: "dump sql tables in a database\n",
+	Long: `
+Dump SQL tables of a cockroach database.
+`,
+	Args: cobra.ExactArgs(1),
+	RunE: MaybeDecorateGRPCError(runDumpSchema),
+	Deprecated: `cockroach dump only supports --dump-mode=schema in v21.1. 
+Please use SHOW CREATE ALL TABLES moving forward.
+cockroach dump will be completely removed in v21.2.
+For details, see: https://github.com/cockroachdb/cockroach/issues/54040`,
+}
+
+// runDumpSchema replaces cockroach dump <database> --dump-mode=schema
+// for v21.1 and will be removed in v21.2.
+// TODO(richardjcai): Remove this in v21.2.
```

Comment:
no need for this todo if we don't merge this to master :)

but also no need to remove it from this PR